### ISSUE TITLE
Duplicated error message

### ISSUE THREAD

**Issue by Tobbe** [dotansimha/graphql-code-generator#7546]
**Describe the bug**
In RedwoodJS we generate types programatically. When there are no files with gql stuff to parse we get a duplicated error message:

```
gen | Generating TypeScript definitions and GraphQL schemas...
gen |
gen |       Unable to find any GraphQL type definitions for the following pointers:
gen |
gen |           - ./web/src/**/!(*.d).{ts,tsx,js,jsx}
gen |
gen |
gen |       Unable to find any GraphQL type definitions for the following pointers:
gen |
gen |           - ./web/src/**/!(*.d).{ts,tsx,js,jsx}
gen |
gen |
gen | Error: Could not generate GraphQL type definitions (web)
gen |
gen | 14 files generated
```

It would look much better for our users if you only output the error message once.

**To Reproduce**

I tried to reproduce the error message in codesandbox, but I couldn't get it to look exactly the same, I get much more output there. But I hope it's close enough to be useful for you. This is what I get when I run `node script.js`

```
sandbox@sse-sandbox-2yriny:/sandbox$ node script.js
(node:1139) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

      Unable to find any GraphQL type definitions for the following pointers:

          - ./web/src/**/!(*.d).{ts,tsx,js,jsx}


      Unable to find any GraphQL type definitions for the following pointers:

          - ./web/src/**/!(*.d).{ts,tsx,js,jsx}

node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

AggregateError: Something went wrong
```

https://codesandbox.io/s/agitated-heyrovsky-2yriny?file=/script.js

1. My GraphQL schema:

See codesandbox link

2. My GraphQL operations:

See codesandbox link

3. My `codegen.yml` config file:

```js
const codegenConfig = {
  schema: "schema.graphql",
  config: {
    namingConvention: "keep",
    // to allow camelCased query names
    scalars: {
      // We need these, otherwise these scalars are mapped to any
      // @TODO is there a way we can use scalars defined in packages/graphql-server/src/rootSchema.ts
      BigInt: "number",
      DateTime: "string",
      Date: "string",
      JSON: "Record<string, unknown>",
      JSONObject: "Record<string, unknown>",
      Time: "string"
    },
    omitOperationSuffix: true // prevent type names being PetQueryQuery, RW generators already append Query/Mutation/etc
  },
  generates: {
    "/Users/tobbe/tmp/typegen/web/types/graphql.d.ts": {
      documents: "./web/src/**/!(*.d).{ts,tsx,js,jsx}",
      plugins: [{ typescript: { enumsAsTypes: true } }, "typescript-operations"]
    }
  },
  silent: false,
  errorsOnly: true
};
```

(Also see codesandbox link)

**Expected behavior**

I expect the error to only show once.

**Environment:**

- OS: MacOS 12
- `@graphql-codegen/cli`: 2.6.1
- NodeJS: 14.19.0

**Additional context**

Issue and draft PR on our side:
https://github.com/redwoodjs/redwood/issues/4393
https://github.com/redwoodjs/redwood/pull/4491

**Comment by Tobbe** [comment]
If I switch this to `false` instead I only get one error as expected https://github.com/dotansimha/graphql-code-generator/blob/cb9adeb962cee5a19064d73f6f21fe2ea488d575/packages/graphql-codegen-cli/src/codegen.ts#L379

But I guess the fix isn't as easy as that. You probably have a good reason for stoping early 🙂

**Comment by charlypoly** [comment]
Hi @Tobbe,

First, sorry for the enormous delay in answering your issue.
I reproduced the duplicated errors on both CLI and script scenarios.
A PR is pending for the CLI scenario: #8120 

For your custom scripts, the duplicated errors come from your script implementation that does not provide a "catch" handler to `cli.generate(/* ... */).then()`.
To avoid the errors to "bubble up" and be printed twice, ensure to pass a proper failure callback as follow:|


```js
cli.generate(codegenConfig, true).then(
  (data) => {
    console.log("data", data);
  },
  () => Promise.resolve() // not providing this callback will trigger a nodejs "triggerUncaughtException"
);```

**Comment by charlypoly** [comment]
@Tobbe the CLI duplicated error messages bug is fixed in `@graphql-codegen/cli@2.9.1` 📦

**Comment by Tobbe** [comment]
Thanks @charlypoly! We've since moved to using `import { codegen } from '@graphql-codegen/core'`. But great to get this fixed nonetheless 🙂
